airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/check/DestinationChecker.kt

```diff
@@ -32,5 +45,48 @@ interface DestinationChecker<C : DestinationConfiguration> {
         )
 
     fun check(config: C)
-    fun cleanup() {}
+    fun cleanup(config: C) {}
+}
+
+class DestinationCheckerSync<C : DestinationConfiguration>(
+    val catalog: DestinationCatalog,
+    pipe: PipedOutputStream,
+    private val writeOperation: WriteOperation,
+    private val cleaner: CheckCleaner<C>,
+) : DestinationChecker<C> {
```

Comment:
Is this class intended to replace the interface?

Reply:
talked offline - tl;dr we should eventually replace the interface, yes. Roughly
* existing connectors need to switch to the new DestinationCheckerSync implementation + delete their existing checker implementations
* for any weird connectors that need to do additional validation, we can have DestinationCheckerSync accept some sort of `() -> Unit`-ish thing, which we execute after the test sync
* and then we can kill the interface + replace all references with just DestinationCheckerSync